pkg/registry/servicecatalog/broker/validation.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
First, I'll move the validation to wherever it's supposed to be. That's `pkg/apis/servicecatalog/validation` as I understand it. I agree that `pkg/api/validation` is not the right way to go.

I don't want to spend time relitigating everything, but I don't want to drop my complete lack of understanding. I want to document this process. I don't think any of the conventions are stupid, I just don't understand them.

What specifically am I supposed to look at in that package? At first couple of glances it seems to be very 'pod' specific code. Specifically for reading pod specs from files and urls.

I also don't want to seem like a five year old, but I need some answers that go a couple 'whys' deep on just about every point you are trying to make. What's the best way to have this discussion upstream? File an issue? File PRs with changes? We need to write down 'why' somewhere. I don't think github threads are the right place.